csharp/ql/src/experimental/Security Features/backdoor/PotentialTimeBomb.ql

```diff
@@ -0,0 +1,152 @@
+/**
+ * @name Potential Timebomb
+ * @description Flow from a file last modification date (very likely implant installation time) and an offset to condition statement (the trigger)
+ * @kind problem
```

Comment:
I suggest turning this into a `path-problem` to better verify the results. In order for this to work, you have to manually extend the `edges` relation:

```ql
query predicate nodes = PathGraph::nodes/3;

query predicate edges(DataFlow::PathNode a, DataFlow::PathNode b) {
  PathGraph::edges(a, b)
  or
  exists(
    FlowsFromGetLastWriteTimeConfigToTimeSpanArithmeticCallable conf1,
    FlowsFromTimeSpanArithmeticToTimeComparisonCallable conf2
  |
    conf1 = a.getConfiguration() and
    conf1.isSink(a.getNode()) and
    conf2 = b.getConfiguration() and
    b.isSource()
  )
  or
  exists(
    FlowsFromTimeSpanArithmeticToTimeComparisonCallable conf1,
    FlowsFromTimeComparisonCallableToSelectionStatementCondition conf2
  |
    conf1 = a.getConfiguration() and
    conf1.isSink(a.getNode()) and
    conf2 = b.getConfiguration() and
    b.isSource()
  )
}
```

Reply:
Working on it.

Reply:
I checked in a `path-problem` version. Please let me know if there is anything I should change. Thanks

Reply:
I don't see the manually added `edges` predicate above? This is needed in order to bind the edges from the 3 configurations together.

Reply:
I didn't noticed the code yesterday. Let me try adding it. Thanks.